Makefile

```diff
@@ -325,7 +341,24 @@ kustomize-crds: output.init $(KUSTOMIZE) $(YQ)
 	XDG_CONFIG_HOME=$(PWD)/package $(KUSTOMIZE) build --enable-alpha-plugins $(OUTPUT_DIR)/package/kustomize -o $(OUTPUT_DIR)/package/crds.yaml || $(FAIL)
 	@$(OK) Kustomizing CRDs.
 
+.PHONY: kustomize-crds
+
 checkout-to-old-api:
 	CHECKOUT_RELEASE_VERSION=$(CHECKOUT_RELEASE_VERSION) hack/check-duplicate.sh
 
-.PHONY: kustomize-crds
+ifeq ($(RUN_BUILDTAGGER),true)
+lint.init: build-lint-cache
+lint.done: delete-build-tags
+
+build-lint-cache: $(GOLANGCILINT)
+	@$(INFO) Running golangci-lint with the analysis cache building phase.
+	@(BUILDTAGGER_DOWNLOAD_URL=$(BUILDTAGGER_DOWNLOAD_URL) ./scripts/tag.sh && \
+	(([[ "${SKIP_LINTER_ANALYSIS}" == "true" ]] && $(OK) "Skipping analysis cache build phase because it's already been populated") && \
+	[[ "${SKIP_LINTER_ANALYSIS}" == "true" ]] || $(GOLANGCILINT) run -v --build-tags account,configregistry,configprovider,linter_run -v --concurrency 1 --disable-all --exclude '.*')) || $(FAIL)
+	@$(OK) Running golangci-lint with the analysis cache building phase.
+
+delete-build-tags:
```

Comment:
As far as I understand, this last operation allows us to remove the tags we added after the lint phase. In this context, the relevant tags are only modified while the linter runs. This is important so that the local env is not affected.

Reply:
Yes, correct. We run the `buildtagger` once more to remove the build tags. Another option would be to copy the code base to a staging folder and generate the build tags on that temporary location so that we don't need to remove the tags. But I preferred this approach because we would need to be careful about what gets copied into the temporary location. I've also updated the developer experience section that warns about the in-place modification during the `make lint` run. If this disturbs other developers, then we can consider linting on a repository copy.